tests/algorithms/test_algorithm_resumption.py

```diff
@@ -101,157 +26,81 @@
         [42, "1ep", "ep{epoch}-rank{rank}", "ep3-rank{rank}", "ep5-rank{rank}"],  # test save at epoch end
     ],
 )
-@pytest.mark.parametrize(
-    "model_name",
-    [pytest.param("default"), pytest.param("resnet50_synthetic", marks=pytest.mark.daily)])
-@pytest.mark.parametrize("algorithm", algorithms_registry.items(), ids=algorithms_registry.keys())
+@pytest.mark.parametrize("algorithm", get_algorithm_registry().keys())
 def test_algorithm_resumption(
-    device_hparams: DeviceHparams,
-    save_interval: str,
+    algorithm: str,
+    device,
+    seed: Optional[int],
+    save_interval: int,
     save_filename: str,
     resume_file: str,
     final_checkpoint: str,
-    seed: Optional[int],
-    model_name: str,
-    algorithm: Tuple[str, Type[AlgorithmHparams]],
-    dummy_model_hparams: ModelHparams,
-    dummy_train_dataset_hparams: DatasetHparams,
-    dummy_val_dataset_hparams: DatasetHparams,
     tmpdir: py.path.local,
 ):
-    """strategy:
-    - train five epochs. capture checkpoints after `checkpoint_interval` and ep5.
-    - create a new trainer from the `checkpoint_interval` checkpoint, and train until end. checkpoint again.
-    - assert that the checkpoint from the new trainer at the end is the same as the checkpoint from the first trainer at the end.
-    """
-
-    algo_name, algo_hparams = algorithm
-
-    if algo_name in xfail_list:
-        pytest.xfail(f"{algo_name} known to fail resumption test")
-
-    if algo_name in skiplist[model_name].keys():
-        pytest.skip(skiplist[model_name][algo_name])
-
-    max_epochs = "5ep"
-    subset_num_batches = 5
-
-    trainer_hparams = TrainerHparams(algorithms=[],
-                                     seed=seed,
-                                     optimizer=AdamHparams(),
-                                     schedulers=[ExponentialSchedulerHparams(gamma=0.9)],
```

Comment:
If you're using the default scheduler hparams, just make sure that gamma is >> 0.1. If gamma is sufficiently small, the learning rate can decay so quickly that the model stops being updated, causing the checkpointing tests to pass (possibly) erroneously.

Reply:
good suggestion! I added a control test to make sure that the model weights are indeed being updated.